Add documentation note for missing cocotb-config (#1564)
cocotb/cocotb: documentation/source/quickstart.rst

    git clone https://github.com/cocotb/cocotb
    pip3 install -e ./cocotb

.. note::

    After installation, you should be able to execute ``cocotb-config``.

>     If it is not found, you need to append to the ``PATH`` environment variable
>     so that it contains an entry for your Python installation.

Reviewer: ```suggestion
    If it is not found, you need to append its location to the ``PATH`` environment variable.
    This may happen when you use the ``--user`` option to ``pip``, in which case the location is documented `here <https://docs.python.org/3/library/site.html#site.USER_BASE>`_.
```

>     This may happen when you use the ``--user`` option to ``pip``, in which case the location is documented `here <python:site.USER_BASE>`_.

Reviewer 2: Perhaps...
```suggestion
    This may happen when you use the ``--user`` option to ``pip``, in which case the location is documented :data:`here <python:site.USER_BASE>`_.
```

>     This may happen when you use the ``--user`` option to ``pip``, in which case the location is documented :ref:`here <python:inst-alt-install-user>`_.

Reviewer 2: Hmm, still not right. Either

```suggestion
    This may happen when you use the ``--user`` option to ``pip``, in which case the location is documented :ref:`here <python:inst-alt-install-user>`.
```
or perhaps
```suggestion
    This may happen when you use the ``--user`` option to ``pip``, in which case the location is documented `here <python:inst-alt-install-user>`_.
```
Maybe we should stop guessing and read the intersphinx docs.

Author: Should work now (works locally for me)

Reviewer: Works: https://cocotb--1564.org.readthedocs.build/en/1564/quickstart.html#installation-via-pip

>     This may happen when you use the ``--user`` option to ``pip``, in which case the location is documented `here <https://docs.python.org/3/library/site.html#site.USER_BASE>`_.

Reviewer 2: ```suggestion
    This may happen when you use the ``--user`` option to ``pip``, in which case the location is documented `here <python:site.USER_BASE>`_.
```
Or something, will generate an intersphinx link.
